src/app/user.service.ts

```diff
@@ -10,23 +9,24 @@ export class UserService {
     private http: HttpClient
   ) { }
   
-  register(user: UserRegistration): Observable<string> {
+  public register(user: UserRegistration): Promise<string> {
     return this.http.post(`${environment.apiUrl}/users/register`, user,  {
       responseType: 'text'
-    }); 
+    }).toPromise(); 
   }
 
-  update(user: UserRegistration): Observable<string> {
+  public update(user: UserRegistration): Promise<string> {
```

Comment:
Why is promise preferred? Though promise is preferred only in server.js?

Reply:
Not entirely true. Promise would be mostly preferred for events that emit once.